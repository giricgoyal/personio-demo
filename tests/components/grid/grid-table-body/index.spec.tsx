import { upperCase } from 'lodash'
import React from 'react'
import GridTableBody from 'src/components/grid/grid-table-body'
import { render } from 'tests/test-utils'

describe('components/grid/grid-table-body', () => {
    let container
    const columnDef = [
        {
            title: 'Name',
            propBinding: 'name',
            filter: true,
            sort: true,
            type: 'string',
            format: (val) => upperCase(val),
        },
    ]

    const data = [
        {
            name: 'John',
            dateOfBirth: '2000-01-01',
        },
        {
            name: 'Jane',
            dateOfBirth: '1991-01-01',
        },
        {
            name: 'Ray',
            dateOfBirth: '1995-01-01',
        },
    ]

    beforeEach(() => {
        ;({ container } = render(<GridTableBody columnDef={columnDef} data={data} />))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })
})
