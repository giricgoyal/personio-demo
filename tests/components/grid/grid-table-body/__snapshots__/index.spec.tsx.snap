// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid/grid-table-body should render and match snapshot 1`] = `
<div>
  <div
    class="grid__table__cell"
  >
    John
  </div>
  <div
    class="grid__table__cell"
  >
    Jane
  </div>
  <div
    class="grid__table__cell"
  >
    Ray
  </div>
</div>
`;
